Mathlib/RingTheory/Henselian.lean

```diff
@@ -236,8 +236,12 @@ instance (priority := 100) IsAdicComplete.henselianRing (R : Type*) [CommRing R]
         obtain ⟨k, rfl⟩ := Nat.exists_eq_add_of_le hmn
         clear hmn
         induction' k with k ih
-        · rw [Nat.zero_eq, add_zero]
-        rw [Nat.succ_eq_add_one, ← add_assoc, hc, ← add_zero (c m), sub_eq_add_neg]
+        · rw [add_zero]
+        rw [← add_assoc]
+        -- Adaptation note: nightly-2024-03-11
+        -- I'm not sure why the `erw` is now needed here. It looks like it should work.
+        erw [hc]
```

Comment:
It looks like it has a problem unfolding `c` itself so tries to unify `Nat.rec` and `HSub.hsub`

Reply:
Do you want to expand the comment?

Reply:
Expanded